app/services/audio.js

```diff
@@ -61,8 +59,17 @@ export default Service.extend({
 
   init() {
     this.get('hifi').on('audio-ended', () => this.finishedTrack());
+    Ember.$(window).on('beforeunload', () => {
+      if (this.get('currentAudio')) {
+        this.sendListenAction(this.get('currentAudio'), 'close');
+      }
+    });
     this._super(...arguments);
   },
+  
+  willDestroy() {
+    Ember.$(window).off('beforeunload');
```

Comment:
Are we sure this will always be the only `'beforeunload'` event? What about namespacing it so we don't end up accidentally stepping on something else later. maybe  `'beforeunload.listenAction'`